packages/styletron-react/src/index.tsx

```diff
@@ -24,15 +27,33 @@ import type {
   WithStyleFn,
   WithTransformFn,
   WithWrapperFn,
-} from "./types.js";
+  StyletronProps,
+} from "./types";
 import {
   addDebugMetadata,
   setupDevtoolsExtension,
   DebugEngine,
-} from "./dev-tool.js";
+} from "./dev-tool";
 
-export {DebugEngine};
-export type {StyleObject};
+export type {
+  StandardEngine,
+  StyleObject,
+  Styletron,
+  StyletronComponent,
+  ReducerContainer,
+  AssignmentCommutativeReducerContainer,
+  NonAssignmentCommutativeReducerContainer,
+  StyledFn,
+  WithStyleFn,
+  WithTransformFn,
+  WithWrapperFn,
+  StyletronProps,
+  addDebugMetadata,
+  setupDevtoolsExtension,
+  DebugEngine,
+};
```

Comment:
this is not strictly required but recommended by api-extractor (it will show warning for types used in exports but not exported)

Reply:
My concern is that changes to largely internal details will require major version releases

Reply:
There are some good reasons why they ask developers to export these types: https://api-extractor.com/pages/messages/ae-forgotten-export/

Reply:
Has this been requested by flow users? I would prefer to take a less open approach and add the feature if requested

Reply:
> My concern is that changes to largely internal details will require major version releases

but it does not really change… it requires type to be exported only if it is already exported as part of something, so technically speaking it is already part of public api

however big part of features of api extractor is actually about managing scope of what should be public api, instead we can add `@internal` in jsdoc for some methods to avoid publishing them